diff --git a/airflow/jobs.py b/airflow/jobs.py
index c372daa..ac3a429 100644
--- a/airflow/jobs.py
+++ b/airflow/jobs.py
@@ -1491,11 +1491,12 @@ class SchedulerJob(BaseJob):
                         dag = dagbag.get_dag(dag_id)
                         ti.task = dag.get_task(task_id)
                         ti.handle_failure(msg)
-                    except Exception:
+                    except Exception as e:
                         self.log.error("Cannot load the dag bag to handle failure for %s"
                                        ". Setting task to FAILED without callbacks or "
                                        "retries. Do you have enough resources?", ti)
-                        ti.state = State.FAILED
+                        self.log.error(e)
+                        ti.state = State.UP_FOR_RETRY
                         session.merge(ti)
                         session.commit()
 
diff --git a/airflow/models.py b/airflow/models.py
index d837810..a732c8c 100755
--- a/airflow/models.py
+++ b/airflow/models.py
@@ -3484,7 +3484,8 @@ class DAG(BaseDag, LoggingMixin):
         if end_date:
             tis = tis.filter(TI.execution_date <= end_date)
         if only_failed:
-            tis = tis.filter(TI.state == State.FAILED)
+            tis = tis.filter(or_(TI.state == State.FAILED,
+                                 TI.state == State.UPSTREAM_FAILED))
         if only_running:
             tis = tis.filter(TI.state == State.RUNNING)
 
